simapp/app_di.go

```diff
@@ -111,7 +111,8 @@ func init() {
 // AppConfig returns the default app config.
 func AppConfig() depinject.Config {
 	return depinject.Configs(
-		appConfig, // Alternatively use appconfig.LoadYAML(AppConfigYAML)
+		appConfig,                               // Alternatively use appconfig.LoadYAML(AppConfigYAML)
+		depinject.Provide(ProvideExampleMintFn), // optional: override the mint module's mint function with epoched minting
```

Comment:
personal opinion: I have seen people just copying the app.go without thinking too much about the configuration. With this change, you make the new mint function default. This may cause some unexpected behaviour and risks.

Reply:
I think it's fine if this becomes the new default for new apps, we'll get some good side effects from this. For example, we can stop producing empty blocks